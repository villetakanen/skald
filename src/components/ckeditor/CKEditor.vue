<template>
  <div style="margin:-8px">
    <ckeditor :editor="editor" v-model="editorData" :config="editorConfig"></ckeditor>
    ---- <br/>
    <div v-html="editorData"></div>
  </div>
</template>

<script>
import Vue from 'vue'
import CKEditor from '@ckeditor/ckeditor5-vue'
import ClassicEditor from '@ckeditor/ckeditor5-build-classic'
import { defineComponent, reactive, toRefs } from '@vue/composition-api'

Vue.use(CKEditor)

export default defineComponent({
  setup () {
    const editorSetup = reactive({
      editor: ClassicEditor,
      editorData: '<p>Content of the editor.</p>',
      editorConfig: {
        // The configuration of the editor.
        fontColor: '#333'
      }
    })

    return { ...toRefs(editorSetup) }
  }
})
</script>

<style lang="sass">
@import '@/styles/ckeditor.scss'
</style>
