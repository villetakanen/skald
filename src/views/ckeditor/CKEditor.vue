<template>
  <v-container>
     <v-row>
        <v-col>
          <TabTitle
            :sub="site.name"
            :sublink="`/v/${site.siteid}`"
            :topic="page.name"
            :link="`/v/${site.siteid}/${page.pageid}`"/>
        </v-col>
      </v-row>
      <v-row>
        <v-col>
          <Loading
            center
            class="ma-4"
            v-if="meta.loading"/>
        </v-col>
      </v-row>
      <v-row v-if="!meta.loading">
      <v-col cols="12" md="8">
        <CKEditorCard/>
      </v-col>
      <v-col cols="12" md="4">
        <PageMetadataCard/>
        <v-card class="my-4">
          <v-card-title>Debug</v-card-title>
          <v-card-text>
            <p>siteid: [{{ siteid }}]</p>
            <p>pageid: [{{ pageid }}]</p>
          </v-card-text>
        </v-card>
      </v-col>
    </v-row>
  </v-container>
</template>

<script lang="ts">
import { defineComponent, ref, onMounted, watch, ComputedRef } from '@vue/composition-api'
import { useParams } from '@/lib/useParams'
import CKEditorCard from '@/components/ckeditorcard/CKEditorCard.vue'
import PageMetadataCard from '@/components/ckeditorcard/PageMetadataCard.vue'
import { usePage } from '@/lib/usePage'
import { useSite } from '@/lib/useSite'
import Loading from '@/components/Loading.vue'
import TabTitle from '@/components/app/TabTitle.vue'

interface PageRoute {
  siteid:string,
  pageid?:string
}

export default defineComponent({
  components: {
    CKEditorCard,
    PageMetadataCard,
    Loading,
    TabTitle
  },
  setup (props) {
    const siteid = ref('')
    const pageid = ref('')
    const { page, meta } = usePage()
    const { site } = useSite()

    /**
     * reactive update of all components of the view,
     * based on siteid and pageid? values of route.
     * If no pageid is give, we use the siteid as pageid.
     *
     * @param p The Vue router `route.params` value from `useParams()`
     */
    function setPage (p:Object) {
      const pageRoute = p as PageRoute
      siteid.value = pageRoute.siteid
      if (pageRoute.pageid) pageid.value = pageRoute.pageid
      else pageid.value = siteid.value
    }

    // useParams, to get the site and page id's
    const params = useParams()
    // reload/remount etc
    onMounted(() => { setPage(params.value) })
    // route changed
    watch(params, setPage)

    return { siteid, pageid, site, page, meta }
  }
})
</script>
