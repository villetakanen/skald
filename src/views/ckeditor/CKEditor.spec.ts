import { shallowMount } from '@vue/test-utils'
import CKEditor from './CKEditor.vue'

describe('Mounted App', () => {
  it('renders correctly', () => {
    const wrapper = shallowMount(CKEditor)
    expect(wrapper.element).toMatchSnapshot()
  })
})
