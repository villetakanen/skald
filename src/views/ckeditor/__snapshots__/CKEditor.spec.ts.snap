// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mounted App renders correctly 1`] = `
<v-container-stub
  tag="div"
>
  <v-row-stub
    tag="div"
  >
    <v-col-stub
      cols="8"
      tag="div"
    >
      <v-card-stub
        loaderheight="4"
        tag="div"
      >
        <v-toolbar-stub
          extensionheight="48"
          src=""
          tag="header"
        >
          <v-toolbar-title-stub>
            CKEditor alpha
          </v-toolbar-title-stub>
        </v-toolbar-stub>
         
        <v-card-text-stub>
          <ckeditor-stub />
        </v-card-text-stub>
      </v-card-stub>
    </v-col-stub>
     
    <v-col-stub
      cols="4"
      tag="div"
    >
      <v-card-stub
        loaderheight="4"
        tag="div"
      >
        <v-card-title-stub>
          Debug
        </v-card-title-stub>
         
        <v-card-text-stub>
          <p>
            siteid: []
          </p>
           
          <p>
            pageid: []
          </p>
        </v-card-text-stub>
      </v-card-stub>
    </v-col-stub>
  </v-row-stub>
</v-container-stub>
`;
